conf/gulp-tasks/bundle/bundleabstractfactory.js

```diff
@@ -0,0 +1,188 @@
+const fs = require('fs');
+
+const { dest } = require('gulp');
+
+const rollup = require('gulp-rollup-lightweight');
+const babel = require('rollup-plugin-babel');
+const resolve = require('rollup-plugin-node-resolve');
+const commonjs = require('rollup-plugin-commonjs');
+const insert = require('rollup-plugin-insert');
+
+const source = require('vinyl-source-stream');
+const replace = require('gulp-replace');
+
+// An enum describing the different types of SDK bundle.
+const BundleType = {
+  MODERN: 'modern',
+  LEGACY_IIFE: 'legacy-iife',
+  LEGACY_UMD: 'legacy-umd'
+};
+Object.freeze(BundleType);
+
+/**
+ * A class providing factories for building different types of SDK bundle.
+ */
+class BundleAbstractFactory {
+  constructor () {
+    this._namespace = 'ANSWERS';
+  }
+
+  /**
+   * Creates a factory for building a specific type of SDK bundle.
+   *
+   * @param {BundleType} bundleType The type of SDK bundle to build.
+   * @returns {Function} Factory method for building an SDK bundle of the
+   *                     the specified type.
+   */
+  create (bundleType) {
```

Comment:
Would it be better just to have individual methods for these, instead of a switch statement + enum?

Reply:
In the Factory design pattern, there's usually a single entry point, like this one. The method takes in a type and spits out the correct version of whatever it is the factory is building. One of the ideas behind this is that it really simplifies the interface for the caller. There's a single method only, they give it a type, it gives them the right object of that type. They don't need to know which particular individual method to call. In this way, the Factory is a bit like the facade design pattern.